tests/test_other.py

```diff
@@ -8246,6 +8246,29 @@ def test_native_link_error_message(self):
     err = run_process([PYTHON, EMCC, 'hello_world.o', '-o', 'hello_world.js'], stdout=PIPE, stderr=PIPE, check=False).stderr
     self.assertContained('hello_world.o is not a valid input', err)
 
+  # Tests that we should a clear error on TOTAL_MEMORY not being enough for static initialization + stack
+  def test_clear_error_on_massive_static_data(self):
+    with open('src.cpp', 'w') as f:
+      f.write('''
+        #include <stdio.h>
+        double muchData[] = {
+      ''')
+      for i in range(50 * 1000):
+        f.write('1.2, 2.3, 3.4, 4.5, 5.6, 6.7, 7.8, 8.9, 9.9,\n')
```

Comment:
Why bother with this huge test?  Why not just create little static data and set TOTAL_MEMORY to a small value?

Reply:
"Tests that we should a" -> "Tests that we give a"